### ISSUE TITLE
[HOLD for payment 2024-07-22] [HOLD for payment 2024-07-17] [HOLD for payment 2024-07-10] [$250] [One expense reports] Update the UI for Held transactions in Single Transaction Report Preview / MoneyRequestAction

### ISSUE THREAD

**Issue by NikkiWines** [Expensify/App#39989]
Slack discussion [here](https://expensify.slack.com/archives/C05RECHFBEW/p1712113359149339)

### Goal 

Expand upon the new UI pattern introduced as part of the [Wave 5/6: Update how we display / report expenses](https://docs.google.com/document/d/1hKK7emsQt4iDvP8Ohvk5aCc8gNdRelyxvbxDkXZxNRA/edit#heading=h.vrxpcdav3nmt) doc to include reports where all transactions that are on  `HOLD` (including reports with only one transaction where we show only one expense) using the copy "All requests were put on hold. Review the comments for next steps."

*Note 1:* that a report could be in two states simultaneously: the receipt could be scanning and the transaction(s) on the report could be on hold 

*Note 2:* One-transaction reports currently do not show an option to hold an expense, but this is easily rectified by including the logic from [here](https://github.com/Expensify/App/blob/4e13dada426c067e2945fbd2bd06ddd73c0b8409/src/components/MoneyRequestHeader.tsx#L113-L136) in `src/components/MoneyReportHeader.tsx` 
<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~017ed767635e9fa640</li>
        <li>Upwork Job ID: 1777965895318855680</li>
        <li>Last Price Increase: 2024-05-10</li>
<li>Automatic offers: </li>
<ul>
<li>dukenv0307 | Reviewer | 0</li>
<li>tienifr | Contributor | 0</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @JmillsExpensify</details>

**Comment by NikkiWines** [comment]
Adding design so we can get some mocks of how this should ideally look 🙇

**Comment by shawnborton** [comment]
> Note 1: that a report could be in two states simultaneously: the receipt could be scanning and the transaction(s) on the report could be on hold

My understanding was that this isn't possible for a one-expense report: you can't hold a scanning receipt. Is that right @trjExpensify @JmillsExpensify ?

**Comment by shawnborton** [comment]
I believe @dragnoir is in the process of fixing that. @dragnoir can you provide an update?

**Comment by shawnborton** [comment]
In terms of the banner style, I posted to Slack but I think it might look something like this: 
![CleanShot 2024-04-10 at 19 26 13@2x](https://github.com/Expensify/App/assets/2319350/de5c0985-50e9-418a-988e-528358f49d9c)

**Comment by JmillsExpensify** [comment]
Can't think of a reason why we'd block holding a scanning receipt, though I'm not super passionate.

**Comment by shawnborton** [comment]
Ah, you said the opposite over [here](https://github.com/Expensify/App/issues/37536#issuecomment-1977234505):
![CleanShot 2024-04-10 at 20 06 34@2x](https://github.com/Expensify/App/assets/2319350/c2a4d6ed-cec4-4a9f-9d8a-f215de7412ef)

**Comment by shawnborton** [comment]
What's the latest here?

**Comment by trjExpensify** [comment]
Bump @dragnoir ^^

**Comment by trjExpensify** [comment]
I'm going to move this into `[One expense reports]` to track for August!

**Comment by NikkiWines** [comment]
Hmm, yeah this issue was intended to be specifically for the one-transaction view. Let's make this external (since we have a mockup [here](https://github.com/Expensify/App/issues/39989#issuecomment-2048103780)) and get this updated!

**Comment by JmillsExpensify** [comment]
Contributor assigned and waiting on PR.

**Comment by trjExpensify** [comment]
>When we put the transaction in one transaction expense report on hold, should we keep the next steps banner? If so, there would be 2 banners at a time.

No, the intention is that there's 1 banner as illustrated [here](https://github.com/Expensify/App/issues/39989#issuecomment-2048103780). That would be the same for a one expense report with a held request, or a multi-expense report where all requests on the report are held. 

>Automatically submit the report in the BE/Show the next steps banner?

No, we shouldn't automatically submit the report if all requests on a draft `open` report are held. We should just show the banner illustrated [here](https://github.com/Expensify/App/issues/39989#issuecomment-2048103780).

**Comment by shawnborton** [comment]
Agree with Tom - thanks for clarifying!

**Comment by trjExpensify** [comment]
We apply it to multi-expense reports when all expenses are held.

**Comment by blimpich** [comment]
Ended up reverting the PR associated with this issue. See [this comment on the PR](https://github.com/Expensify/App/pull/42374#issuecomment-2154737349). See where we decided to do this internally [here](https://expensify.slack.com/archives/C01GTK53T8Q/p1717717225546139?thread_ts=1717699373.036439&cid=C01GTK53T8Q).

The deploy blockers that caused us to revert were [this one](https://github.com/Expensify/App/issues/43207) and [this one](https://github.com/Expensify/App/issues/43203). We need to redo this PR and make sure those don't happen.

**Comment by trjExpensify** [comment]
@tienifr did you open that PR?

**Comment by trjExpensify** [comment]
>When all expenses within an expense report are on hold (same with one-transaction expense report), should we show submit and settlement (approve/pay) button in report preview and report header?

- An expense report can be submitted with held requests, so if the report is in the open state they can `Submit`. 
- An expense report can be approved/paid with held requests, we just display a modal when clicked to ask if they want to unhold any held requests to approve/pay them all, or only the expenses on the report that are unheld.

>should we display any message in the report preview indicating the hold status of the expense report?

When an expense report contains held requests, the report preview component in the parent chat has a red dot in the header. It doesn't contain a message currently, and I don't think we need to add that here. 

<img width="823" alt="image" src="https://github.com/Expensify/App/assets/16232057/1a25cca1-181a-481c-a57e-5b14e235e3ff">

**Comment by JmillsExpensify** [comment]
Agree with everything @trjExpensify mentioned.

**Comment by JmillsExpensify** [comment]
@alexpensify are we paying this PR via your issue or this one?

**Comment by alexpensify** [comment]
I started the payment process in my GH. We can combine the two since it was one PR.

**Comment by JmillsExpensify** [comment]
Ok cool, so then I'll just close this one out. Someone please comment if you disagree.
